tests/test_decorators.py

```diff
@@ -30,6 +31,8 @@
 
 api = hug.API(__name__)
 module = sys.modules[__name__]
+test_directory = os.path.dirname(os.path.realpath(__file__))
```

Comment:
Ah, I think my use case is a bit different from what you have in mind. I often run py.test with a single test file, so that when I am developing a certain feature I don't have to execute the whole test suite. When doing so my CWD could be anything really, often it is the ``tests`` directory itself. 

So if you prefer autoenv, maybe it's possible to make an autoenv function which accepts arbitrary parameters, ``cd``s into the root directory, runs ``py.test`` from there and passes all parameters to it.